src/app/page.tsx

```diff
@@ -1,24 +1,61 @@
-import { CategoriesExpandableList } from "@/components/CategoriesExpandableList";
-import { CategoriesMenu } from "@/components/CategoriesMenu";
-import Header from "@/components/Header";
+"use client";
+
+import { useState } from "react";
+import dynamic from "next/dynamic";
+
+import CategoriesExpandableList from "@/components/CategoriesExpandableList";
 import ListItemCard from "@/components/ListItemCard";
+import { Modal } from "flowbite-react";
+import PreviewItemsAddedSkeleton from "@/components/PreviewItemsAdded/PreviewItemAddedSkeleton";
+
+const PreviewItemAdded = dynamic(
```

Comment:
Preview Item is imported dynamically as it is not required on page load, it can be `lazy loaded` only once it's displayed on the screen, and to ease in, we use a skeleton for it